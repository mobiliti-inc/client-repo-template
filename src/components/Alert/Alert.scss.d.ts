// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'container': string;
  'container--closable': string;
  'container--hide': string;
  'fades-in-out': string;
  'icon': string;
  'icon--close': string;
  'sub-title': string;
  'title': string;
}
declare var cssExports: CssExports;
export = cssExports;
