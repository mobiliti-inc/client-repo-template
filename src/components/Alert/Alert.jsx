import React from 'react';
import classNames from 'classnames';

import { iconClose } from '../../assets/icons';

import styles from './Alert.scss';

const Alert = ({
	show,
	title,
	subTitle,
	children,
	closable,
	onClose,
}) => (
	<div styleName={classNames('container', { 'container--hide': !show, 'container--closable': closable })}>
		{closable && <div styleName="icon" onClick={onClose} role="presentation"><img styleName="icon--close" src={iconClose} alt="Close" /></div>}
		{title && <span styleName="title">{title}</span>}
		{subTitle && <span styleName="sub-title">{subTitle}</span>}
		{children && children}
	</div>
);


Alert.propTypes = {
	show: PropTypes.bool.isRequired,
	title: PropTypes.string,
	subTitle: PropTypes.string,
	children: PropTypes.node,
	closable: PropTypes.bool,
	onClose: PropTypes.func,
};

Alert.defaultProps = {
	title: '',
	subTitle: '',
	children: null,
	closable: true,
	onClose: () => {},
};

export default CSSModules(Alert, styles, { allowMultiple: true });
