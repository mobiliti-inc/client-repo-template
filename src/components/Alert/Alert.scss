@import "../../styles/_variables.scss";

.container {
	position: fixed;
	top: 0;
	width: 50%;
	z-index: 200;
	cursor: pointer;
	transition: all 0.5s ease;
	border-radius: 5px;
	// Content display properties
	background-color: $white;
	margin: 0;
	padding: 20px;
	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
	transform: translate(0, 75%);
	display: flex;
	flex-direction: column;
	// Traansition properties
	visibility: visible;
	opacity: 1;

	@media (min-width: $laptop-width) {
		width: 40%;
	}

	&--hide {
		display: none;
		visibility: hidden;
		opacity: 0;
		transition-duration: 1s;
		transition-property: opacity;
	}

	.title {
		color: $blue;
		font-family: $futura;
		font-size: 24px;
		font-weight: bold;
		line-height: 31px;
		margin: 0;
		width: 100%;
	}

	&--closable {
		.title {
			margin: 20px 0 0 0;
		}
	}

	.sub-title {
		color: $mid-gray;
		font-family: $hanken-sans;
		font-size: 15px;
		line-height: 19px;
		margin: 0;
		padding-top: 5px;
		width: 100%;
		font-weight: 200;
	}

	.icon {
		display: flex;
		justify-content: center;
		height: 15px;
		width: 15px;
		position: absolute;
		top: 20px;
		right: 20px;

		&--close {
			height: 15px;
			width: 15px;
		}
	}
}