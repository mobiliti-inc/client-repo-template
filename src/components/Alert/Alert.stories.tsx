import { storiesOf } from "@storybook/react";
import { text, boolean } from "@storybook/addon-knobs";
import * as React from "react";
import { Alert } from "../";

storiesOf("Alert", module).add("default", () => (
	<Alert show={boolean('show', true)} title={text('title', "Default title")} subTitle={text('subTitle', "Default subtitle")} closable={boolean("closable", true)} />
));
