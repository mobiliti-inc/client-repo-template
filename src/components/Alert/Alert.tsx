import React from 'react';
import cx from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from "@fortawesome/pro-light-svg-icons";

import './Alert.scss';

interface AlertProps {
	show: boolean;
	title?: string;
	subTitle?: string;
	closable?: boolean;
	onClose?: (...args: any[]) => any;
}

const Alert: React.FC<AlertProps> = (props) => {
	const { show, title, subTitle, children, closable, onClose } = props;
	return (
		<div styleName={cx('container', { 'container--hide': !show, 'container--closable': closable })}>
			{closable && (
				<div styleName="icon" onClick={onClose} role="presentation">
					<div styleName="icon--close"><FontAwesomeIcon icon={faTimes} /></div>
				</div>
			)}
			{title && <span styleName="title">{title}</span>}
			{subTitle && <span styleName="sub-title">{subTitle}</span>}
			{children && children}
		</div>
	);
};

export default Alert;
