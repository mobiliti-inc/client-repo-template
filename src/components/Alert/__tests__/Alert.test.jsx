import Alert from '../Alert';

describe('Alert', () => {
	it('renders', () => {
		const subject = <Alert show scrolling />;
		expect(subject).toMatchSnapshot();
	});
});
