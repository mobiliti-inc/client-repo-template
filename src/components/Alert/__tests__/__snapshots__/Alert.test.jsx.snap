// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders 1`] = `
<Alert
  closable={true}
  onClose={[Function]}
  scrolling={true}
  show={true}
  subTitle=""
  title=""
/>
`;
